pacfish_matlab/@kwave_adapter/kwave_adapter.m

```diff
@@ -1,101 +1,390 @@
+                                obj.position2 = (dist(elem.start_point(2),elem.end_point(2))/2) + min(elem.start_point(2),elem.end_point(2));
+                                 % This finds the distance between the start
+                                % point and end point in the y dimension, halves it and adds
+                                % it to the smaller distance to get
+                                % midpoint
+                            else
+                                 % This sets the position2 variable to the
+                                % lines
+                                % start point in the y dimension
+                                obj.position2 = elem.start_point(2);
+                            end
+                            device_struct.detectors.(index).detector_position = [obj.position1-min_1 obj.position2-min_2];
+                            device_struct.detectors.(index).detector_geometry_type = "LINE"; % They are line elements
+                            if dist(elem.start_point(1),elem.end_point(1)) ~= 0 % Checks whether the distance between the start point/end point in the x direction is not 0 - if it is then it does the same check for the other 2 dimensions
+                                 % Length is determined by checking the
+                                % distance between the start point and end
+                                % point of the line in the x direction
+                                elem.length = dist(elem.start_point(1),elem.end_point(1));
+                                % Width is the line height determined
+                                % earlier
+                                elem.width = obj.lineheight;
+                            elseif dist(elem.start_point(2),elem.end_point(2)) ~= 0
+                                 % Length is determined by checking the
+                                % distance between the start point and end
+                                % point of the line in the y direction
+                                elem.length = dist(elem.start_point(2),elem.end_point(2)); 
+                                % Width is the line height determined
+                                % earlier
+                                elem.width = obj.lineheight;
+                            end
+                        end
+                            % Orientation doesn't exist for lines
+                             % Geometry only requires 2 dimensions here x
+                            % and y as input is 2D
+                            device_struct.detectors.(index).detector_geometry = [elem.length elem.width];
+                end
+                
+                
+                
+                %% FIXME: we need a FOV definition.
+                if any(obj.fov)
+                    device_struct.general.field_of_view = obj.fov;
+                else
+                    device_struct.general.field_of_view = [];
+                end

```

Comment:
Should we maybe consider creating a standardised default definition?
E.g. in #34

Reply:
Just checking, by that you mean we would determine what would be used as the field of view if none is given (aka when else is triggered) ?